crates/dbsp/src/utils/dyn_vec.rs

```diff
@@ -510,7 +554,9 @@ where
         // Deallocate all allocated memory
         let layout = self.layout_for(self.capacity());
         // Safety: The current pointer was allocated with the given layout
-        unsafe { alloc::dealloc(self.as_mut_ptr(), layout) };
+        if self.capacity > 0 {
+            unsafe { alloc::dealloc(self.as_mut_ptr(), layout) };
```

Comment:
It's probably implied by capacity being non-zero but I'd add a debug_assert that self.as_mut_ptr isn't null here since you're calling an unseafe function.